snapcraft/internal/build_providers/_base_provider.py

```diff
@@ -190,16 +193,39 @@ def _push_file(self, *, source: str, destination: str) -> None:
         """Push a file into the instance."""
 
     @abc.abstractmethod
+    def is_mounted(self, target: str) -> bool:
+        """Query if there is a mount at target mount point."""
+
+    @abc.abstractmethod
+    def mount(self, host_source: str, target: str) -> None:
```

Comment:
the model we decided to follow for this base class is that if we do not intend to use these methods outside of the components inheriting from it, they should be private/protected (start with `_`)